pkg/api/copy_test.go

```diff
@@ -32,24 +32,24 @@ import (
 
 func TestDeepCopyApiObjects(t *testing.T) {
 	for i := 0; i < *fuzzIters; i++ {
-		for _, gv := range []unversioned.GroupVersion{testapi.Default.InternalGroupVersion(), *testapi.Default.GroupVersion()} {
-			f := apitesting.FuzzerFor(t, gv.String(), rand.NewSource(rand.Int63()))
-			for kind := range api.Scheme.KnownTypes(gv) {
-				doDeepCopyTest(t, gv.String(), kind, f)
+		for _, version := range []unversioned.GroupVersion{testapi.Default.InternalGroupVersion(), *testapi.Default.GroupVersion()} {
+			f := apitesting.FuzzerFor(t, version.String(), rand.NewSource(rand.Int63()))
```

Comment:
Yes, group should be a domain name for disambiguation (mygroup.mycompany.com).
We should probably make explicit that our groups with no domain following
are in the k8s.io domain or some such.

On Wed, Dec 9, 2015 at 11:28 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/api/copy_test.go
> <https://github.com/kubernetes/kubernetes/pull/18379#discussion_r47190465>
> :
>
> > @@ -32,24 +32,24 @@ import (
> >
> >  func TestDeepCopyApiObjects(t *testing.T) {
> >  	for i := 0; i < *fuzzIters; i++ {
> > -		for _, gv := range []unversioned.GroupVersion{testapi.Default.InternalGroupVersion(), *testapi.Default.GroupVersion()} {
> > -			f := apitesting.FuzzerFor(t, gv.String(), rand.NewSource(rand.Int63()))
> > -			for kind := range api.Scheme.KnownTypes(gv) {
> > -				doDeepCopyTest(t, gv.String(), kind, f)
> > +		for _, version := range []unversioned.GroupVersion{testapi.Default.InternalGroupVersion(), *testapi.Default.GroupVersion()} {
> > +			f := apitesting.FuzzerFor(t, version.String(), rand.NewSource(rand.Int63()))
>
>
>    - GroupVersion is more like a Package if I were to take Java naming
>    conventions. This also make sense to me since if we start to support
>    third-party API groups, I would want to add the other fields to a
>    Package like vendor for example.
>
> I think somewhere smarterclayton, lavalamp, and bgrant0607 has agreed
> vendor should be part of the GroupVersion.Group, and use "." to delimiter
> vendor and group.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18379/files#r47190465>.
>